src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationRow.tsx

```diff
@@ -395,3 +341,75 @@ const ButtonHolder = styled('div')`
     margin-left: ${space(0.5)};
   }
 `;
+
+type ActionButtonsProps = {
+  org: Organization;
+  app: SentryApp;
+
+  showDashboard: boolean;
+  showPublish: boolean;
+  showDelete: boolean;
+  onPublish?: () => void;
+  onDelete?: (app: SentryApp) => void;
+  // If you want to disable the publish or delete buttons, pass in a reason to display to the user in a tooltip
+  disablePublishReason?: string;
+  disableDeleteReason?: string;
+};
+const ActionButtons = ({
+  org,
+  app,
+  showDashboard,
+  showPublish,
+  showDelete,
+  onPublish,
+  onDelete,
+  disablePublishReason,
+  disableDeleteReason,
+}: ActionButtonsProps) => {
+  const appDashboardButton = (
+    <StyledButton
+      size="small"
+      icon="icon-stats"
+      to={`/settings/${org.slug}/developer-settings/${app.slug}/dashboard`}
+    >
+      {t('Dashboard')}
+    </StyledButton>
+  );
+
+  const publishRequestButton = (
+    <StyledButton
+      disabled={!!disablePublishReason}
+      title={disablePublishReason}
+      icon="icon-upgrade"
+      size="small"
+      onClick={onPublish}
+    >
+      {t('Publish')}
+    </StyledButton>
+  );
+
+  const deleteConfirmMessage = t(
+    `Deleting ${app.slug} will also delete any and all of its installations. \
+     This is a permanent action. Do you wish to continue?`
+  );
+  const deleteButton = disableDeleteReason ? (
+    <StyledButton disabled title={disableDeleteReason} size="small" icon="icon-trash" />
+  ) : (
+    <ConfirmDelete
+      message={deleteConfirmMessage}
+      confirmInput={app.slug}
+      priority="danger"
+      onConfirm={() => onDelete && onDelete(app)}
+    >
+      <StyledButton size="small" icon="icon-trash" />
+    </ConfirmDelete>
+  );
+
+  return (
+    <ButtonHolder>
+      {showDashboard && appDashboardButton}
```

Comment:
This is very minor in terms of performance, but I think this will result in an additional function call for creating the app dashboard button element. If it were inlined the conditional would short circuit and it would not happen. Alternatively, we could move the evaluation of `showDashboard` to where we define `appDashboardButton`.